src/test/libcephfs/test.cc

```diff
@@ -984,12 +984,23 @@ TEST(LibCephFS, PreadvPwritev) {
   struct iovec iovin[2];
   struct iovec iovout[2];
   ssize_t nwritten, nread;
+  char *out0, *out1, *in0, *in1; 
 
+  out0 = new char[str0.size()];
+  out1 = new char[str1.size()];
+  in0 = new char[str0.size()];
+  in1 = new char[str1.size()];
+
+  iovout[0].iov_base = out0;
```

Comment:
Well, I ever thought about this, but I searched the difference between c_str() and data() and learned that c_str() is always null terminated while data() is not. Regarding PreadvPwritev(), I use a string to do the test, so c_str() is selected here. Hence, strcpy() is necessary.